src/tir/usmp/analysis/extract_buffer_info.cc

```diff
@@ -203,29 +214,40 @@ void BufferInfoExtractor::RecordAllocateNodeInfo(const AllocateNode* op) {
   auto size_bytes = CalculateExtentsSize(op);
   // We only statically memory plan only allocates with known
   // compile time sizes.
-  if (size_bytes.defined() &&
-      allocate_var_to_stmt_map_.find(op->buffer_var) == allocate_var_to_stmt_map_.end()) {
-    // By default, the core compiler is assumed to attach the a default pool to each allocate.
-    ICHECK(op->annotations.count(kPoolCandidatesAllocateAttr))
-        << "Every statically sized allocate node needs an pool candidate attribute";
-    auto pool_candidates = Downcast<Array<PoolInfo>>(op->annotations[kPoolCandidatesAllocateAttr]);
-
-    // TODO(@manupa-arm): improve the error when the responsible component for attaching a single
-    // pool is added
-    ICHECK(pool_candidates.size() > 0)
-        << "The core compiler should at least attach a single PoolInfo. If there were no "
-           "user-given arguments for memory pools, the default behaviour is a single size "
-           "un-restricted pool is assigned";
-    PrimFunc func = scope_stack_.top().func;
-    Optional<Target> tgt = func->GetAttr<Target>(tvm::attr::kTarget);
-    ICHECK(tgt) << "There should not be any PrimFuncs without a target attached by now";
-    auto workspace_alignment =
-        tgt.value()->GetAttr<Integer>("workspace-byte-alignment").value_or(16);
-    auto buffer_info = BufferInfo(GetUniqueBufferName(op->buffer_var->name_hint), size_bytes,
-                                  pool_candidates, workspace_alignment);
-    auto allocate = GetRef<Allocate>(op);
-    allocate_var_to_stmt_map_.Set(op->buffer_var, allocate);
-    buffer_info_map_.Set(buffer_info, allocate);
+  if (size_bytes.defined()) {
```

Comment:
I think these changes need rationalizing in the commit message as I can't immediately see how they relate to the greedy algorithm.

Reply:
https://github.com/apache/tvm/pull/9214/commits/26fffb0fbfac1558bef905dee1817ba2e5eeabdf.

It is mentioned in "a commit" message. I ll capture this into the main message of the PR.
I appreciate that it is truly not part of the greedy algorithm, however getting too small value after the greedy allocation is what actually made me find the issue. 

Therefore, I hope updating the main PR commit message should help here.